### ISSUE TITLE
Convert to `null` and `unserializable`

### ISSUE THREAD

**Issue by simon-id** [DataDog/dd-native-appsec-js#43]
As stated in [this issue](https://github.com/DataDog/libddwaf/issues/14) libddwaf cannot represent nulls or unserializables.
Once it is implemented, we should use those instead of the empty maps currently used in:
- src/convert.cpp#L87
- src/convert.cpp#L116

**Comment by simon-id** [comment]
New types landed:

### New object types

Alongside the schema extraction preprocessor, two new types have been introduced to ensure a more accurate and complete schema can be produced. These are `float` and `null`, the former for completeness of the numerical types and the latter for its semantical value which, in the context of schema extraction, differs from invalid in that it signifies a null value rather than an unknown type.

Library bindings with a mirrored definition of `ddwaf_object` should now include the `f64` field, of type `double`, in the value union:

```cpp
...
    union
    {
        const char* stringValue;
        uint64_t uintValue;
        int64_t intValue;
        ddwaf_object* array;
        bool boolean;
        double f64;
    };
...
```
This new field breaks with the naming convention of the current `ddwaf_object` definition, however it matches the naming convention of the future `ddwaf_object` definition which will be included in version 2.0.

Similarly, those bindings mirroring the enum types should also include the two new types:
```cpp
...
    // 64-bit float (or double) type
    DDWAF_OBJ_FLOAT    = 1 << 6,
    // Null type, only used for its semantical value
    DDWAF_OBJ_NULL    = 1 << 7,
...
```

These new types can now be created with their corresponding `ddwaf_object` constructors:
```cpp
ddwaf_object* ddwaf_object_null(ddwaf_object *object);
ddwaf_object* ddwaf_object_float(ddwaf_object *object, double value);
```

And finally, float values can also be accessed with the corresponding getter:
```cpp
double ddwaf_object_get_float(const ddwaf_object *object);
```

**Comment by simon-id** [comment]
https://github.com/DataDog/libddwaf/compare/1.10.0...1.14.0
